src/main/java/edu/umich/its/cpm/MigrationController.java

```diff
@@ -646,44 +648,311 @@ private void JSON_response(HttpServletResponse response, String jsonValue,
 			log.error("Missing box integration parameters");
 			return null;
 		}
-		String remoteUserEmail = request.getRemoteUser();
-		
-		if (BoxUtils.getBoxAccessToken() == null)
-		{
+		String remoteUserEmail = request.getRemoteUser() + "@umich.edu";
+
+		if (BoxUtils.getBoxAccessToken() == null) {
 			// go to Box authentication screen
 			// get access token and refresh token and store locally
 			BoxUtils.authenticate(boxAPIUrl, boxClientId, boxClientRedirectUrl,
 					remoteUserEmail, response);
-		}
-		else
-		{
+		} else {
 			// get box folders json
 			return BoxUtils.getBoxFolders(boxClientId, boxClientSecret);
 		}
 		return null;
 	}
 
-	@RequestMapping("/box/folders_authorized")
+	@RequestMapping("/authorized")
 	@Produces(MediaType.APPLICATION_JSON)
-	public List<HashMap<String, String>> getBoxFoldersAuthorized(
-			HttpServletRequest request) {
+	public void getBoxAuthzTokens(HttpServletRequest request) {
 
 		String boxClientId = env.getProperty(BOX_CLIENT_ID);
 		String boxClientSecret = env.getProperty(BOX_CLIENT_SECRET);
 		String boxAPIUrl = env.getProperty(BOX_API_URL);
 		String boxTokenUrl = env.getProperty(BOX_TOKEN_URL);
 		log.info("token url=" + boxTokenUrl);
-		
+
 		String rv = "";
-		// get the authCode
-		String authCode = BoxUtils.getAuthCodeFromBoxCallback(request, boxClientId, boxClientSecret, boxTokenUrl);
+		// get the authCode, 
+		// and get access token and refresh token subsequently
+		BoxUtils.getAuthCodeFromBoxCallback(request,
+				boxClientId, boxClientSecret, boxTokenUrl);
+	}
 
-		if (boxClientId == null || boxClientSecret == null || authCode == null) {
-			log.error("Missing box integration parameters (Box client id, client secrect or authCode) ");
-			return null;
+	/**
+	 * upload resource files into Box folder
+	 * 
+	 * @return
+	 */
+	@RequestMapping("/box/upload")
+	public void boxUpload(HttpServletRequest request,
+			HttpServletResponse response) {
+
+		String boxClientId = env.getProperty(BOX_CLIENT_ID);
+		String boxClientSecret = env.getProperty(BOX_CLIENT_SECRET);
+		String boxClientRedirectUrl = env.getProperty(BOX_CLIENT_REDIRECT_URL);
+		String boxAPIUrl = env.getProperty(BOX_API_URL);
+		// need to have all Box app configurations
+		if (boxClientId == null || boxClientSecret == null) {
+			log.error("Missing Box integration parameters (Box client id, client secret)");
+		}
+		String remoteUserEmail = request.getRemoteUser();
```

Comment:
in handleBoxFolders() you add "@umich.edu" to the remoteUser (which Kyle rightly questions in regard to friend accounts) - perhaps the better thing to do is get the user email from LDAP instead of appending (or not appending) "@umich.edu"?